core-util/Array.h

```diff
@@ -45,6 +46,12 @@ class Array {
     /** Create a new array
       */
     Array(): _head(NULL) {
+        _grow_capacity = 0;
+        _alloc_traits = {0};
+        _alignment = MBED_UTIL_POOL_ALLOC_DEFAULT_ALIGN;
+        _capacity = 0;
+        _elements = 0;
+        _element_size = 0;
```

Comment:
In this case it is a matter of style, initializer list or might be better to use c++11 initializers :question: